### ISSUE TITLE
Guide to export custom yolo models

### ISSUE THREAD

**Issue by hardikdava** [luxonis/tools#51]
Hello, 

- is there any guide on how to convert custom yolo version (other than yolov5, yolov6, yolov7, yolov8)? 
- which layers should be modified in order to do on-device-decoding?

**Comment by hardikdava** [comment]
@tersekmatija , thanks for your reply.

I want to run damo-yolo detection model. The onnx model has 2 output nodes i.e. boxes (in the form of xyxy) and scores (shape = (number of classes, 1)). Since the boxes are already in the form of xyxy, I think I may not require decoding. But I have to find best class and then pass it to nms. The numpy operations are as follows.

```
output = self.model.run(None, {self.model.get_inputs()[0].name: net_image})   ## onnx prediction
scores = output[0][0]
bboxes = output[1][0]     

confidences = np.max(scores, axis=1)
valid_mask = confidences > conf_thresh
boxes = bboxes[valid_mask]
scores = scores[valid_mask]
class_ids = np.argmax(scores, axis=1)
confidences = confidences[valid_mask]

valid_boxes = non_maximum_suppression(boxes, confidences, iou_thresh)

```

**Comment by tersekmatija** [comment]
Yeah, that's already wrapped by their API. You can look [here](https://github.com/tinyvision/DAMO-YOLO/blob/master/damo/base_models/heads/zero_head.py#LL66C11-L66C11). I'd have to look deeper and investigate to see if it matches any of the current versions but will not have the time to do so anytime soon. It also seems the accuracy is not that much better and latency is measured on T4 which can deviate from OAK-D.

We could perhaps look into exposing the NMS node? CC: @themarpe

**Comment by hardikdava** [comment]
@tersekmatija actually, if you expose NMS is a good idea. That will bring easy to extend custom models fully compatible to run inside device itself.

**Comment by klemen1999** [comment]
Exposing the NMS in native DAI node is currently not on the short-term roadmap so I'm closing this issue. If this will be implemented down the line though I'll let you know.
Klemen
